### PR TITLE
SUS-1636: Restrict and cleanup WallNotifications controllers

### PR DESCRIPTION
**Pull request by mszabo-wikia**
- removed checkTopic method and forced Wall js to use MW API instead
- removed redundant nirvana dispatch through UpdateWiki
- made WallNotificationsControllerBase into WikiaService subclass
- escaped i18n

https://wikia-inc.atlassian.net/browse/SUS-1636

### CODE REVIEW

File: extensions/wikia/Wall/js/MessageTopic.js [hunk 1]
```diff
@@ -48,21 +48,21 @@
...
 		handleRemove: function(e) {
 			var topic = $(e.target).closest('.topic');
 			this.removeSelection(topic.data('article-title'));
 			topic.remove();
 		},
-		
+
 		getTopics: function() {
 			var topics = [];
 			for(var selection in this.selectedEntries) {
```
Comment (WikiaTech) [inline extensions/wikia/Wall/js/MessageTopic.js:60]: MAJOR Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 13 issues
* MAJOR 5 major
* MINOR 6 minor
* INFO 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [MessageTopic.js#L69](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/Wall/js/MessageTopic.js#L69): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [MessageTopic.js#L130](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/Wall/js/MessageTopic.js#L130): Remove the unused function parameter "data". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [WallNotificationControllerBase.class.php#L194](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php#L194): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. MAJOR [WallNotificationControllerBase.class.php#L226](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php#L226): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MINOR [WallNotificationControllerBase.class.php#L55](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php#L55): Define a constant instead of duplicating this literal "grouped" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallNotificationControllerBase.class.php#L77](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php#L77): Define a constant instead of duplicating this literal "displayname" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallNotificationControllerBase.class.php#L222](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php#L222): Immediately return this expression instead of assigning it to the temporary variable "$msg". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [WallNotificationControllerBase.class.php#L232](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php#L232): Define a constant instead of duplicating this literal "http://" 5 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallNotificationsExternalController.class.php#L57](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/WallNotifications/WallNotificationsExternalController.class.php#L57): Define a constant instead of duplicating this literal "status" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallNotificationsExternalController.class.php#L70](https://github.com/Wikia/app/blob/e5c9ae1a4942185e7009e5a14023c6b65cd51d33/extensions/wikia/WallNotifications/WallNotificationsExternalController.class.php#L70): Define a constant instead of duplicating this literal "unread" 5 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php [hunk 2]
```diff
@@ -276,6 +247,28 @@ protected function fixNotificationURL( $url ) {
 	}
 
 	/**
+	 * @param string $title
+	 * @return string
+	 */
+	protected function shortenTitle( string $title ): string {
+		return $this->getWallHelper()->shortenText( $title, static::NOTIFICATION_TITLE_LIMIT );
```
Comment (macbre) [inline extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php]: `shortenText` method can be made static in `WallHelper` class, saving object creation and reducing the number of methods (`getWallHelper`). We can also get rid of `$app->wg->Lang` there.

**Review by macbre — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 37 issues
* MAJOR 8 major
* MINOR 20 minor
* INFO 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [WallHelper.class.php#L6](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/Wall/WallHelper.class.php#L6): Class "WallHelper" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [WallHelper.class.php#L260](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/Wall/WallHelper.class.php#L260): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. MAJOR [WallHelper.class.php#L549](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/Wall/WallHelper.class.php#L549): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [MessageTopic.js#L69](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/Wall/js/MessageTopic.js#L69): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. MAJOR [MessageTopic.js#L130](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/Wall/js/MessageTopic.js#L130): Remove the unused function parameter "data". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MAJOR [WallNotificationControllerBase.class.php#L191](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php#L191): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. MAJOR [WallNotificationControllerBase.class.php#L223](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php#L223): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MINOR [WallHelper.class.php#L20](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/Wall/WallHelper.class.php#L20): Define a constant instead of duplicating this literal "title" 7 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallHelper.class.php#L130](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/Wall/WallHelper.class.php#L130): Define a constant instead of duplicating this literal "timestamp" 5 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [WallHelper.class.php#L138](https://github.com/Wikia/app/blob/af47b4d2ab060d45081e7c1f173306824fbaabf5/extensions/wikia/Wall/WallHelper.class.php#L138): Define a constant instead of duplicating this literal "wall-title" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Review by macbre — APPROVED** [review]

**Pull request merged**